proto/pbconfigentry/config_entry.proto

```diff
@@ -13,6 +13,10 @@ enum Kind {
   KindIngressGateway = 3;
   KindServiceIntentions = 4;
   KindServiceDefaults = 5;
+  KindInlineCertificate = 6;
+  KindAPIGateway = 7;
+  KindHTTPRoute = 8;
+  KindTCPRoute = 9;
```

Comment:
Last thing, we'll also need `BoundAPIGateway` stuff here, since `pbconfigentry` stuff is used in `proxycfg` and the event publishers for actually marshaling the config entries into protobuf format for sending over gRPC.